cardano-api/test/Test/Cardano/Api/TxBody.hs

```diff
@@ -0,0 +1,285 @@
+normalizeMetadata = \case
+  TxMetadataInEra _ (TxMetadata m) | null m -> TxMetadataNone
+  other                                     -> other
+
+-- | Unify empty and None.
+-- Upgrade script versions (see Upgrading scripts section).
+-- Stabilize order of scripts sorting them by language.
+normalizeAuxScripts :: TxAuxScripts era -> TxAuxScripts era
+normalizeAuxScripts = \case
+  TxAuxScripts _       []      -> TxAuxScriptsNone
+  TxAuxScripts support scripts ->
+    -- sorting uses script versions, hence sort after upgrade
+    TxAuxScripts support $
+    sortOn languageOfScriptInEra $ map upgradeScriptInEra scripts
+  other -> other
+
+languageOfScriptInEra :: ScriptInEra era -> AnyScriptLanguage
+languageOfScriptInEra (ScriptInEra lang _) =
+  AnyScriptLanguage $ languageOfScriptLanguageInEra lang
+
+-- | Unify empty and None.
+normalizeWithdrawals :: TxWithdrawals ViewTx era -> TxWithdrawals ViewTx era
+normalizeWithdrawals = \case
+  TxWithdrawals _ [] -> TxWithdrawalsNone
+  other              -> other
+
+-- | Unify empty and None.
+normalizeCertificates :: TxCertificates ViewTx era -> TxCertificates ViewTx era
+normalizeCertificates = \case
+  TxCertificates _ [] _ -> TxCertificatesNone
+  other                 -> other
+
+-- | Unify empty and None.
+normalizeMintValue :: TxMintValue ViewTx era -> TxMintValue ViewTx era
+normalizeMintValue = \case
+  TxMintValue _ v _ | v == mempty -> TxMintNone
+  other                           -> other
+
+
+-- * Ugrading scripts
+--
+-- The instruction set from V1 may be used as V2,
+-- and we can't determine the language version from the transaction.
+-- When the user uses V1 instructions,
+-- we don't know if they used the V1 tag or the V2 tag.
+-- So it's safe to drop this information.
+
+upgradeScriptInEra :: ScriptInEra era -> ScriptInEra era
+upgradeScriptInEra = \case
+  ScriptInEra SimpleScriptV1InAllegra script ->
+    ScriptInEra SimpleScriptV2InAllegra $ upgradeScript script
+  ScriptInEra SimpleScriptV1InMary script ->
+    ScriptInEra SimpleScriptV2InMary $ upgradeScript script
+  ScriptInEra SimpleScriptV1InAlonzo script ->
+    ScriptInEra SimpleScriptV2InAlonzo $ upgradeScript script
+  other -> other
+
+upgradeScript :: Script SimpleScriptV1 -> Script SimpleScriptV2
+upgradeScript (SimpleScript SimpleScriptV1 script) =
+  SimpleScript SimpleScriptV2 $ upgradeSimpleScript script
+
+upgradeSimpleScript ::
+  SimpleScript SimpleScriptV1 -> SimpleScript SimpleScriptV2
+upgradeSimpleScript = \case
+  RequireSignature hash -> RequireSignature hash
+  RequireAllOf scripts  -> RequireAllOf $ map upgradeSimpleScript scripts
+  RequireAnyOf scripts  -> RequireAnyOf $ map upgradeSimpleScript scripts
+  RequireMOf n scripts  -> RequireMOf n $ map upgradeSimpleScript scripts
+
+
+-- * Review: Change 'TxBodyContent' “phase” to 'ViewTx'
+--
+-- Basically, it's just dropping witness information.
+
+review :: TxBodyContent BuildTx era -> TxBodyContent ViewTx era
```

Comment:
`review` isn't a very indicative name. We should rename this to something like "toTxBodyContentView".

Reply:
I was looking for a prefix for a group of functions. What about "view" and "build"?
```
viewBodyContent
viewTxIn
viewWithdrawals
viewCertificates
viewMintValue
buildBodyContent
buildTxIn
buildWithdrawals
buildCertificates
buildMintValue
```

Reply:
Because I think prefixes serve to group the functions better than suffixes.

Reply:
"view" is better imo :+1: